version: '3.8'

services:
  db:
    image: postgres:17-alpine
    container_name: postgres_db_doctor_record_service
    environment:
      POSTGRES_PASSWORD: admin123
      POSTGRES_DB: medi_track
    ports:
      - "5435:5432"
    volumes:
      - postgres_data:/var/lib/postgresql@17/data:rw
    networks:
      - my-app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    container_name: mt_doctor_record_service
    environment:
      DB_USER: postgres
      DB_PASSWORD: admin123
      DB_NAME: medi_track
      DB_HOST: host.docker.internal
      DB_PORT: 5435
    ports:
      - "8004:8084"
    depends_on: 
      db:
        condition: service_healthy
    networks:
      - my-app-network

volumes:
  postgres_data:

networks:
  my-app-network:
    external: true